### PR TITLE
Add flakehell python linter (#3295)

### PR DESCRIPTION
**Pull request by a666**
Adds flakehell.

For the moment it is just a blatant copy of the flake8 linter.

### CODE REVIEW

File: ale_linters/python/flakehell.vim [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    return l:cwd
+endfunction
+
+function! ale_linters#python#flakehell#GetCommand(buffer, version) abort
+    let l:executable = ale_linters#python#flakehell#GetExecutable(a:buffer)
+
+    if (l:executable =~? 'pipenv\|poetry$')
+        let l:exec_args = ' run flakehell'
+    elseif (l:executable is? 'python')
+        let l:exec_args = ' -m flakehell'
```
Comment (a666) [inline ale_linters/python/flakehell.vim:74]: This seemed to make more sense that appending it in the options.

**Review by hsanson — CHANGES_REQUESTED** [review]
Impressive PR, thanks.  One test on windows is failing, once that is fixed it looks good to merge:

```
    ( 7/17) [EXECUTE] (X) ['flakehell', ['flakehell --version', 'flakehell lint --format=default --stdin-display-name %s -']] should be equal to ['C:\testplugin\test\test-files\python\with_virtualenv\env\Scripts\flakehell', ['C:\testplugin\test\test-files\python\with_virtualenv\env\Scripts\flakehell --version', 'C:\testplugin\test\test-files\python\with_virtualenv\env\Scripts\flakehell lint --format=default --stdin-display-name %s -']]
 ```

**Comment by a666** [comment]
Thanks.
I'm unsure about how to fix that test, it is on parity with the (non-failing) flake8 test:
```diff
-Execute(The flake8 callbacks should detect virtualenv directories):
+Execute(The flakehell callbacks should detect virtualenv directories):
   silent execute 'file ' . fnameescape(g:dir . '/../test-files/python/with_virtualenv/subdir/foo/bar.py')

   let b:executable = ale#path#Simplify(
-  \ g:dir . '/../test-files/python/with_virtualenv/env/' . b:bin_dir . '/flake8'
+  \ g:dir . '/../test-files/python/with_virtualenv/env/' . b:bin_dir . '/flakehell'
   \)

   AssertLinter b:executable, [
   \ ale#Escape(b:executable) . ' --version',
   \ ale#Escape(b:executable)
+  \   . ' lint'
   \   . ' --format=default'
   \   . ' --stdin-display-name %s -',
   \]
```
And I don't see anything in the code that's done any differently with respect to flake8.
Am I missing something?

**Comment by a666** [comment]
Final test now passes.

**Review by hsanson — APPROVED** [review]
Thanks for the contribution.

**Pull request merged**